### ISSUE TITLE
Ack skips over files it should not (bad regexp in is_searchable)

### ISSUE THREAD

**Issue by mernst** [beyondgrep/ack1#101]
The ack documentation states:

```
   Some files will never be selected by ack-grep, even with -a, including:

   * Backup files: Files ending with ~, or #*#
   * Coredumps: Files matching core.\d+
```

and this is hard-coded in the subroutine is_searchable.

However, ack skips over files that it should not.  This may be due, at
least in part, to an error in is_searchable's hard-coded regular
expression.

To reproduce the problem, run the following commands:

```
% cd
% mkdir ack-test-issearchable
% cd ack-test-issearchable
% mkdir 'a#b'
% touch 'a#b/c#'
% touch 'a#b/d#e'
% touch 'f#g#'
% touch 'h#i#j'
% ack --noenv -f -u
a#b/c#
a#b/d#e
h#i#j
f#g#
% ack --noenv -f -a
a#b/c#
a#b/d#e
h#i#j
% ack --noenv -f
% 
```

The output for "ack --noenv -f -u" is correct.  The output for the other
two commands should be identical to it, I believe.

**Comment by mernst** [comment]
Bump.
This is not a hypothetical issue; ack's mishandling of file names makes it unusable for use cases that are important to me.

**Comment by petdance** [comment]
This is fixed in 1.92 that I am releasing today.
